### PR TITLE
Add regendoc runs for fixture docs

### PR DESCRIPTION
**Pull request by matthewhughes934**
* Fix various typos in fixture docs

* Add basic emaillib for tests in fixture.rst

    This will be used to power regendoc runs for later tests

* Add regendoc runs for emaillib tests in fixture

    Also update these tests ensure they pass.

    This addresses #8213

-----

I've run `regendoc` locally to ensure the added tests pass, but the resulting diff was rather large (and included some bad whitespace) so I've left it out. What might be best here? I was thinking of just commiting the results of the added tests.

### CODE REVIEW

**Review by bluetech — APPROVED** [review]
The changes look good to me!

**Review by nicoddemus — APPROVED** [review]
LGTM, thanks a lot @matthewhughes934!

> I've run regendoc locally to ensure the added tests pass, but the resulting diff was rather large (and included some bad whitespace) so I've left it out. What might be best here? I was thinking of just commiting the results of the added tests.

I'm surprise there were many changes... did you run on Linux using Python 3.8?

File: doc/en/fixture.rst [hunk 1]
```diff
@@ -844,17 +846,48 @@ Once the test is finished, pytest will go back down the list of fixtures, but in
...
+            other.inbox.append(email)
+
+        def clear_mailbox(self):
+            self.inbox.clear()
+
+
+    class Email:
+        def __init__(self, subject, body):
+            self.body = body
+            self.subject = subject
```
[the remark this answers is not in the document]
Comment (matthewhughes934) [reply doc/en/fixture.rst]: :+1: done a51c72f

**Comment by matthewhughes934** [comment]
> I'm surprise there were many changes... did you run on Linux using Python 3.8?

Ah, I was running on 3.9. Diff is reasonable after a run on 3.8: 10dd3db

**Comment by nicoddemus** [comment]
Awesome, thanks a lot!

Normally we squash the commits before merging, but I noticed you have some commits with `!fixup`... would you like to rebase the history yourself before we merge, or can I go ahead and hit `Squash` here on GitHub?

**Comment by matthewhughes934** [comment]
> Awesome, thanks a lot!
> 
> Normally we squash the commits before merging, but I noticed you have some commits with `!fixup`... would you like to rebase the history yourself before we merge, or can I go ahead and hit `Squash` here on GitHub?

:+1: rebased

**Pull request merged**